webapp/src/blocks.tsx

```diff
@@ -891,14 +892,77 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         this.parent.fireResize();
     }
 
-    listenToBlockErrorChanges(handlerKey: string, handler: (errors: pxtblockly.BlockDiagnostic[]) => void) {
-        this.errorChangesListeners[handlerKey] = handler;
+    onExceptionDetected(exception: pxsim.DebuggerBreakpointMessage) {
+        const displayInfo: ErrorDisplayInfo = this.getDisplayInfoForException(exception);
+        this.errors = [displayInfo];
+    }
+
+    private onBlockErrorChanges(errors: ErrorDisplayInfo[]) {
+        this.errors = errors;
+    }
+
+    private getDisplayInfoForBlockError(blockId: string, message: string): ErrorDisplayInfo {
+        return {
+            message: message,
+            onClick: () => {
+                // Block may already be highlighted, so we want to set highlighted directly to true
+                // as opposed to using `highlightBlock`, which toggles.
+                const block = this.editor.getBlockById(blockId);
+                if (!block) {
+                    return;
+                }
+
+                block.setHighlighted(true);
+                this.editor.centerOnBlock(blockId);
```

Comment:
```suggestion
                this.editor.centerOnBlock(blockId, true);
```

you need to pass true here because without it the editor will center on the block and all blocks that are connected after it